packages/auth/amplify_auth_cognito_dart/lib/src/sdk/sdk_exception.dart

```diff
@@ -83,49 +81,41 @@ export 'cognito_identity_provider.dart'
 /// service SDKs.
 @internal
 Exception transformSdkException(Exception e) {
+  if (e is! SmithyException) {
+    return e;
+  }
+  final message = e.message ?? e.toString();
+  // Some exceptions are returned as non-Lambda exceptions even though they
+  // orginated in user-defined lambdas.
+  if (LambdaException.isLambdaException(message) ||
```

Comment:
I realize this was already part of the code (and so maybe we can fix later), but I don't know if we want to collapse these flavors of Lambda Exception into one type, since according to the [Cognito docs](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_InitiateAuth.html#API_InitiateAuth_Errors) they have different connotations (even if they aren't super clear).

Reply:
I like the idea of letting them drill down with an `is` check.